sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -417,6 +417,57 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
           Context const& context) const override;
     };
 
+    /**
+     * @brief Helper class for creating scopes for the token request context.
+     *
+     */
+    struct TokenScopes
```

Comment:
Per our design patterns this should be a class.

Reply:
Guidelines says:

  -  YOU SHOULD declare classes with only public members using class-key struct

https://azure.github.io/azure-sdk/cpp_implementation.html#cpp-design-naming-classes-no-struct-keyword

This is a class with only a public function

Reply:
public members != public function

For consistency, we use structs when it only has public members like some of our options or models types. Types with functions should be classes.

If a type only contains fields (and likely will remain that way), use struct. Otherwise, use class.

Reply:
What's the benefit from doing 

```cpp
class Foo {
 public:
    static void Foo();
}
```

vs

```cpp
struct Foo {
    static void Foo();
}
```

I don't think the guideline is looking for creating an `only properties` for struct, but looking looking to avoid adding the `public` keyword when everything inside the class is public (being redundant).

Reply:
I'd prefer class. To me, class is a (potential) black box, while struct is (potential) "record". And this is more of a black box, or at least it may evolve that way.

Reply:
We use struct for similar cases like `StringExtensions`, `CaseInsensitiveComparator`.

A container which is not expected to ever hold a private state but only offer public state or public methods.

> To me, class is a (potential) black box,

the black/white box is a concept for how a consumer will interact and see the component. As a consumer, I can't see if the component is using a class or a struct right?  That's only  internal concept.  A consumer will never try to see if he's using a class or a struct to understand usage or how to use a component.

The class vs struct usage should give us at least a tiny benefit of avoiding redundancy.

Reply:
> We use struct for similar cases like `StringExtensions`, `CaseInsensitiveComparator`.

Those are internal and didn't go through the rigorous review process. They shouldn't have been structs.

> The class vs struct usage should give us at least a tiny benefit of avoiding redundancy.

The key benefit to the customer is consistency. The key benefit to us, as developers and contributors, is to not have this debate over and over again on each PR. We should settle and stick to the one approach and avoid having this decision point while coding/catching in code review. That is a large dev velocity benefit to sticking to it consistently :)

Reply:
> I plan to keep bringing things I don't think are right or where we might be building paradigms for not-too-relevant things like struct vs class..

That's great. Please do.
Can you please set something up with the team offline so we can discuss this together and close on it (along with updating any guidelines for clarity).

Reply:
BTW, should've this thing been public, we would almost certainly make it a class, because we would need to delete at least the default constructor, so that no one attempts to create an instance of this class, which will be a step in a wrong direction.
And, we'd make it `final`.

Reply:
why are we spending so much time with pedantic discussions on this irrelevant issue ? 
based on all the information i could find structs and classes in c++ are identical except the default access, you can have methods, inheritance , constructor, destructor etc. 
there is no difference in functionality, capability, speed, build time, etc. 
So then why would we want to put a ban on a benign language feature? 
If it was C ,sure ,there are differences, in C++  .... seems to me it's just an academic discussion, based on preference and nothing else. 
As for consistency , since they are the same thing it is consistent.